django/db/backends/oracle/operations.py

```diff
@@ -84,6 +84,8 @@ def date_extract_sql(self, lookup_type, field_name):
         if lookup_type == 'week_day':
             # TO_CHAR(field, 'D') returns an integer from 1-7, where 1=Sunday.
             return "TO_CHAR(%s, 'D')" % field_name
+        elif lookup_type == 'week':
+            return "WEEKOF(%s)" % field_name
```

Comment:
``WEEKOF`` is not defined in the `Oracle`. You should use ``TO_CHAR`` with ``IW`` param
```diff
--- a/django/db/backends/oracle/operations.py
+++ b/django/db/backends/oracle/operations.py
@@ -84,6 +84,9 @@ WHEN (new.%(col_name)s IS NULL)
         if lookup_type == 'week_day':
             # TO_CHAR(field, 'D') returns an integer from 1-7, where 1=Sunday.
             return "TO_CHAR(%s, 'D')" % field_name
+       elif lookup_type == 'week':
+            # TO_CHAR(field, 'IW') returns an integer from 1-52 or 1-53, week of the year based on the ISO standard
+            return "TO_CHAR(%s, 'IW')" % field_name
         else:
             # http://docs.oracle.com/cd/B19306_01/server.102/b14200/functions050.htm
             return "EXTRACT(%s FROM %s)" % (lookup_type.upper(), field_name)
```